### ISSUE TITLE
Macro can replace existing command

### ISSUE THREAD

**Issue by KS2499** [KS2499/pe#5]
Macro replaces mapping of default commands as well like `:refresh`

While it is mentioned macro will replace already defined commands, it does not mention it will replace commands that are preloaded with the app.

Steps to reproduce:
1. Launch app
2. Add some tasks
3. execute `:macro -l`
4. result = `{}`
This implies no macros are present
5. execute `:macro -a refresh "a hello"`
6. Execute `:refresh`
It will now add a task called `hello` to the list and the functionality of refreshing to show the entire list is lost, only recovered by deleting this macro.

In conclusion, Macro should not be able to replace default commands, or mapping of default commands should be displayed when executing `:macro -l`

**Comment by nus-se-script** [comment]
# Team's Response
First, `macro -l` can only list all macros that exist at the moment the command is executed. In your case, you executed `macro -l` before you added a new macro.

The current implementation is a design choice - we think that the user has to be responsible when using this powerful feature. Letting the user define a macro that can overshadow existing commands gives us more flexibility - let's say, a user wants to set the default of the sort command to sort by deadlines, they can set a macro `s` that maps to  `s -d`. Now, they just need to type `s` to sort by deadlines.

Also, in our UG, we mentioned that the motivation behind the macro features is from bash alias - and in bash, aliases can also overshadow the existing commands (one example is `ls` that maps to `ls --color=auto`). We agree that blocking the users from giving the macro the same name as default commands is a valid option, but that would limit the power of macros too much.

Furthermore, the user can simply delete the macro or even delete the file and it is unlikely that this flaw will affect normal operations of the product.
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.Rejected`]

- [x] I disagree

**Reason for disagreement:** My argument is as follows:
- In the UG it is said `If the macro is already defined, this command will replaced the old mapping with the new one.`
- To check mappings, you execute `You can view all macros and their mappings by keying in :macro -l`
- When you do so when you have not added any mappings, it results in a result of `{}`
- Thus this means I can define macros with the same names as default commands
- I do so and now I lose the ability to execute the default command
- The mapping of the default command is rewritten, however I had no idea the default command had a mapping

The issue is if I had inadvertently assigned a macro that replaces a command I had not used till that point, do not use it for a while, come back and want to use the command as defined in UG but it does the behaviour in my macro definition. Now I am inconvenienced by having to debug why it is not working as described in the UG.

I however accept due to the nature of this issue it is `severity.Low` as mentioned by the dev team. 

P.S. I have a feeling they wanted to accept some bugs at lower severity but lowered the severity and then rejected it, why bother lowering severity in this case

<catcher-end-of-segment><hr>
## :question: Issue severity

Team chose [`severity.Low`]
Originally [`severity.High`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
